### DISCUSSION TITLE
Discord usernames are not always correctly transferred

### DISCUSSION THREAD

**Discussion by yldshv** [pocketbase/pocketbase#4681] [Q&A] [answered]
Hello everyone,

today I came across something that seemed odd to me. I use Discord for authentication in my app, and normally the username in the users collection is set to the Discord username. However, with a new user today, it was set as "users5383". So, I did a bit of research and found that the Discord name of the user begins with a period, like ".name". Since I have a Discord bot that needs to access the username, this is obviously not ideal. Additionally, through the Pocketbase UI, I can't adjust the username to the desired format because a period at the beginning is not supported. Therefore, I created an endpoint for myself where I can change the username from my application -> this works (although I'm not sure if it could lead to further problems later).

**My question: Is there a correct solution for this?**

Thank you and best regards,
Daniel

**Comment by ganigeorgiev** [comment, marked as answer]
The system `username` field format cannot be changed at the moment.
I've already refactored this as part of https://github.com/pocketbase/pocketbase/discussions/4355 and users will have more control over the system fields in the future but it will take time before I can publish it.

> Therefore, I created an endpoint for myself where I can change the username from my application -> this works (although I'm not sure if it could lead to further problems later).

Without a code sample it is unclear what this custom endpoint do so it is difficult to say whether it could cause problems or not. Another alternative that doesn't require custom route could be to create your own `discordUsername` unique `text` field in the `users` collection and populate it with the Discord field in `OnRecordAfterAuthWithOAuth2Request` hook. If you use the JSVM this could look something like:

```js
onRecordAfterAuthWithOAuth2Request((e) => {
    // update the username field with the one from the discord account
    // (note: if you want this do be done only for new records check e.isNewRecord)
    if (e.providerName == "discord") {
        // note: technically you can also update here directly the "username" field since no validations will be performed
        // but it may return an error if the users want to update their record later as it will trigger the default username field validations
        e.record.set("discordUsername", e.oAuth2User.username)
        $app.dao().saveRecord(e.record)
    }
}, "users")
```

**Reply by yldshv** [reply]
Thanks for the fast answer!
Below you can see what the endpoint does -> its just changing the username :D 
Btw i like your approach as well. 

**EDIT: I just saw your edits, soo question already answered. Ty!**

**Do you think this approach will break something else?** (pocketbase wise in my app it will not break anything)
```go
type UserNames struct {
  Old string
  New string
}

var userNames UserNames
err := json.NewDecoder(c.Request().Body).Decode(&userNames)
if err != nil {
  fmt.Println(err)
}

record, err := app.Dao().FindAuthRecordByUsername("users", userNames.Old)
if err != nil {
  return err
}

record.Set("username", userNames.New)

if err := app.Dao().SaveRecord(record); err != nil {
  return err
}
```
